engine/time/src/main/java/io/deephaven/time/calendar/Calendar.java

```diff
@@ -30,6 +28,68 @@ public class Calendar {
     private final String description;
     private final ZoneId timeZone;
 
+    // region Cache
+
+    private static class SummaryData extends ImmutableConcurrentCache.IntKeyedValue {
+        final LocalDate startDate;
+        final LocalDate endDate; // exclusive
+        final List<LocalDate> dates;
+
+        SummaryData(int key, LocalDate startDate, LocalDate endDate, List<LocalDate> dates) {
+            super(key);
+            this.startDate = startDate;
+            this.endDate = endDate;
+            this.dates = dates;
+        }
+
+    }
+
+    private final YearMonthSummaryCache<SummaryData> summaryCache =
+            new YearMonthSummaryCache<>(this::computeMonthSummary, this::computeYearSummary);
+
+    private SummaryData summarize(final int key, final LocalDate startDate, final LocalDate endDate) {
+        LocalDate date = startDate;
+        final ArrayList<LocalDate> dates = new ArrayList<>();
+
+        while (date.isBefore(endDate)) {
+            dates.add(date);
+            date = date.plusDays(1);
+        }
+
+        return new SummaryData(key, startDate, endDate, dates); // end date is exclusive
+    }
+
+    private SummaryData computeMonthSummary(final int yearMonth) {
+        final int year = yearMonth / 100;
+        final int month = yearMonth % 100;
```

Comment:
This can use the `io.deephaven.time.calendar.YearMonthSummaryCache#yearFromMonthKey / monthFromMonthKey`?